### DISCUSSION TITLE
Hireable NPC cost isnt calculated for special NPCs

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#13725] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Hireable NPC logic that calculates the hiring cost doesn't work for custom NPCs.

![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/dbb44fd6-6000-445e-90b7-5a37432986f6)

```xml
  <npcset identifier="huskhireables">
      <npc identifier="hireablehuskscientist" faction="huskcult" >
        <Character speciesname="human" startitemsgiven="false" >
          <job identifier="medicaldoctor">
            <skill identifier="helm" level="5" />
            <skill identifier="weapons" level="15" />
            <skill identifier="medical" level="60" />
            <skill identifier="mechanical" level="10" />
            <skill identifier="electrical" level="10" />
          </job>
          <talents>
            <talent identifier="laresistance" />
            <talent identifier="selfcare" />
            <talent identifier="medicalassistance" />
            <talent identifier="egghead" />
          </talents>
        </Character>
        <ItemSet commonness="1">
          <Item identifier="idcard" equip="true" />
          <Item identifier="headset" equip="true" showpreview="false" />
          <Item identifier="cultistrobeshoodless" equip="true" />
          <Item identifier="healthscanner" equip="true"/>        
          <Item identifier="antibloodloss1"/>
          <Item identifier="calyxanide"/>
          <Item identifier="antidama1"/>
          <Item identifier="huskeggsbasic"/>
        </ItemSet>
      </npc>
      <npc identifier="hireablehusktemplar" faction="huskcult">
        <Character speciesname="human" startitemsgiven="false" >
          <job identifier="securityofficer">
            <skill identifier="helm" level="5" />
            <skill identifier="weapons" level="60" />
            <skill identifier="medical" level="15" />
            <skill identifier="mechanical" level="10" />
            <skill identifier="electrical" level="10" />
          </job>
          <talents>
            <talent identifier="buff" />
            <talent identifier="swole" />
            <talent identifier="stonewall" />
            <talent identifier="crustyseamen" />
          </talents>
        </Character>
        <ItemSet commonness="2">
          <Item identifier="idcard" equip="true" tags="id_outpostsecurity"/>
          <Item identifier="cultistrobes" equip="true"/>
          <Item identifier="bodyarmor" equip="true"/>        
          <Item identifier="divingknife" equip="false"/>
          <Item identifier="handcuffs"/>
          <Item identifier="Shotgun" equip="false">
            <Item identifier="shotgunshell" amount="6"/>
          </Item>
            <Item identifier="shotgunshell" amount="12"/>
          <Item identifier="syringegun" equip="false">
            <Item identifier="chloralhydrate" amount="4"/>
          </Item>
        </ItemSet>
      </npc>
  </npcset>
```

_No response_

### Version

Unstable

**Comment by ubertpendragon** [comment]
As a result you end up with very lame faction hires. Your only customisation options are giving some starting XP to hires.

Some loadouts aren't even the correct one, like husk hireable (pirate bandana? vip clothes?)
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/c06c6668-4841-4514-aa55-6c24105811b0)

**Comment by Regalis11** [comment]
It seems that if the character's skills have been defined manually like this, you also need to define the cost manually, e.g. `<npc identifier="hireablehusktemplar" faction="huskcult" salary="3000">`. There would be no random variance in the cost anyway, because it's based on the skill levels, which in this case are fixed too.

**Reply by ubertpendragon** [reply]
I looked for it but didnt know that was a thing, will try 👍 

You should take a look at Husk/Clown hires they don't have a name defined and their loadout feels... not appropirate
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/7790eef1-16bd-446b-bae4-5d609cf128b5)

**Comment by ubertpendragon** [comment]
> It seems that if the character's skills have been defined manually like this, you also need to define the cost manually, e.g. `<npc identifier="hireablehusktemplar" faction="huskcult" salary="3000">`. There would be no random variance in the cost anyway, because it's based on the skill levels, which in this case are fixed too.

Salary is defined in character and not in identifier, but this works. Unfortunately the reputation doesn't affect the salary

      <Character speciesname="human" startitemsgiven="false" salary="6000" >
      
      
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/e882c600-ff4f-4a85-826b-ebcc9a482ad1)
